syntax = "proto3";

package user;

message WalletUpdateRequest {
  string user_id = 1;
  double amount_change = 2;
  string reason = 3;
}

message WalletResponse {
  double new_balance = 1;
  bool success = 2;
}

service UserService {
  rpc UpdateWallet(WalletUpdateRequest) returns (WalletResponse);
}
